### DISCUSSION TITLE
Persist last selected value from options picker.

### DISCUSSION THREAD

**Discussion by magickaito** [Budibase/budibase#5975] [Help] [answered]
I have an Options picker, that shows a list of "Report" titles from a Report table.  

I want to persist the selection, so that last selected report is always automatically selected.

I am able to persist the numeric report ID in onChange action, by writing {{ Field Value }} into State.lastSelectedReportID.

But when I set default value of the picker as {{ State.lastSelectedReportID }}, the numeric value is displayed in the picker: 

![Screenshot 2022-05-19 at 4 00 27 PM](https://user-images.githubusercontent.com/1195573/169243512-eb22921f-7bd9-4451-a05c-fd8f6e83cda7.jpg)

Nothing is preselected. How do I auto select the report element where report ID = State.lastSelectedReportID?

**Comment by melohagan** [comment, marked as answer]
Hi @magickaito 

This is possibly a data typing issue. Handlebars always return string types, and I'm assuming the ID is a Number type?

Instead of saving the handlebar ` {{ Field Value }} ` to state, try using the JavaScript instead. Something like:

`return $("Field Value");`

**Reply by magickaito** [reply]
yes! changing to javascript solves the problem. Thanks.
